src/components/app/data/hooks/useIsCourseRunUpgradable.js

```diff
@@ -0,0 +1,35 @@
+import { useQuery } from '@tanstack/react-query';
+
+import { queryCanUpgradeWithLearnerCredit } from '../queries';
+import useEnterpriseCustomer from './useEnterpriseCustomer';
+
+/**
+ * Determines whether the given content identifier is contained within the enterprise customer's catalogs.
+ * @returns {Types.UseQueryResult}
+ */
+export default function useIsCourseRunUpgradable(contentIdentifiers, queryOptions = {}) {
+  const { data: enterpriseCustomer } = useEnterpriseCustomer();
+  const {
+    select, refetchOnMount, refetchOnWindowFocus, ...queryOptionsRest
+  } = queryOptions;
+  return useQuery({
+    ...queryCanUpgradeWithLearnerCredit(enterpriseCustomer.uuid, contentIdentifiers),
+    ...queryOptionsRest,
+    refetchOnMount: false,
+    refetchOnWindowFocus: false,
+    select: (data) => {
+      if (data.length === 0) {
+        return [];
+      }
+      return data.flatMap((canRedeem) => {
+        if (enterpriseCustomer.hideCourseOriginalPrice) {
+          return {
+            ...canRedeem,
+            listPrice: null,
+          };
+        }
```

Comment:
nit/thought/suggestion: I'm somewhat inclined to leave any conditional display rendering of price up to the JSX as opposed to a hook that provides data since the conditional is more related to the UI layer than data layer. Food for thought (opinion very loosely held lol)

Reply:
After giving this some thought, I've decided to go with removing this conditional from the hook to the UI layer mainly because of the possibility of a `null` list price coming back from the API. There would still need to be another condition in UI layer regardless, so consolidating the logic in the UI layer makes the most sense 👍🏽